solidity/random-beacon/test/RandomBeacon.Callback.test.ts

```diff
@@ -74,26 +63,23 @@ describe("RandomBeacon - Callback", () => {
     testToken = contracts.testToken as TestToken
     callbackContract = contracts.callbackContractStub as CallbackContractStub
     callbackContract1 = contracts.callbackContractStub1 as CallbackContractStub
-
-    await randomBeacon.updateRelayEntryParameters(
-      to1e18(100),
-      relayEntrySubmissionEligibilityDelay,
-      relayEntryHardTimeout,
-      callbackGasLimit
-    )
-    await randomBeacon.updateGroupCreationParameters(
-      groupCreationFrequency,
-      groupLifetime
-    )
   })
 
   describe("requestRelayEntry", () => {
-    beforeEach(async () => {
+    before(async () => {
+      await createSnapshot()
+
       await approveTestToken()
     })
 
+    after(async () => {
```

Comment:
Would it work if we create a snapshot in the `before` block, store the snapshot ID, and then in `afterEach` we would restore a snapshot with a specific ID instead of poping the stack?

Reply:
Nope, as I mentioned in the PR description, `evm_snapshot` and `evm_revert` have some limitations. If you use `evm_revert` with a specific ID, this snapshot is restored but the ID is removed and can't be reused. Also, all IDs created after the removed ID are removed as well. This is why stacking snapshots is the only possible way if we want to use multiple snapshots and fit into our nested test structure.

Reply:
😿 
🤷🏻‍♂️